### ISSUE TITLE
Rule to require that Promise executor function calls resolve

### ISSUE THREAD

**Issue by jroru** [eslint-community/eslint-plugin-promise#438]
### Description

A rule could be created for the Promise constructor executor function to require that the resolve callback is called.

Example of incorrect code for this rule:
```ts
await new Promise(() => {
  setTimeout(() => {}, 2000);
});
```

Example of correct code for this rule:
```ts
await new Promise((resolve) => {
  setTimeout(resolve, 2000);
});
```
